client/state/ui/reducer.js

```diff
@@ -74,6 +75,13 @@ export const isPreviewShowing = createReducer( false, {
 		isShowing !== undefined ? isShowing : state,
 } );
 
+// FIXME: We can't detect window size on the server, so until we have more intelligent detection,
+// use 769, which is just above the general maximum mobile screen width.
+export const viewportWidth = createReducer( 769, {
+	[ VIEWPORT_WIDTH_SET ]: ( state, { width } ) =>
+		! isNaN( width ) ? width : state,
+} );
```

Comment:
is there a reason to use `createReducer()` here? seems like a very simple reducer that doesn't demand the abstraction…

```js
export const viewportWidth = ( state = 769, { type, width } ) =>
	VIEWPORT_WIDTH+SET === type && isNaN( width )
		? width
		: state;
```

Reply:
fun fact @mcsf - I did a test recently and it's hard to beat a switch case

I'm almost tempted to stop using `get()` and start using switches only
https://jsperf.com/proxy-request-origin-matching

switches at least are basic for people to see if they are also a bit verbose